File: task/task-api/local-logger/local-logger.go
package localLogger

import (
	"fmt"
	"github.com/EspressoTrip-v2/concept-go-common/logcodes"
	"github.com/EspressoTrip-v2/concept-go-common/logging"
	"github.com/EspressoTrip-v2/concept-go-common/microservice/microserviceNames"
	"github.com/streadway/amqp"
)

var localLogger *localLoggerConfig

type localLoggerConfig struct {
	serviceName microserviceNames.MicroserviceNames
	logger      *logging.LogPublish
}

func Start(rabbitChannel *amqp.Channel, serviceName microserviceNames.MicroserviceNames) {
	if localLogger == nil {
		logger := localLoggerConfig{
			serviceName: serviceName,
			logger:      logging.NewLogPublish(rabbitChannel, serviceName),
		}
		localLogger = &logger
	}
}

func Log(logCode logcodes.LogCodes, message string, origin string, details string) {
	if localLogger == nil {
		fmt.Printf("[logger:%v]: Has not been started, logs will not be sent", localLogger.serviceName)
	} else {
		localLogger.logger.Log(logCode, message, origin, details)
	}
}

File: employee-dash/employee-dash-api/src/utils/local-logger.ts
import { LogCodes, LogPublisher, MicroServiceNames } from "@espressotrip-org/concept-common";
import amqp from "amqplib";

export class LocalLogger {
    static m_instance: LocalLogger;
    m_logger: LogPublisher;

    private constructor(rabbitChannel: amqp.Channel, microserviceName: MicroServiceNames) {
        this.m_logger = new LogPublisher(rabbitChannel, microserviceName);
    }

    static start(rabbitChannel: amqp.Channel, microserviceName: MicroServiceNames): void {
        if (this.m_instance) return;
        this.m_instance = new LocalLogger(rabbitChannel, microserviceName);
    }

    static log(logCode: LogCodes, message: string, origin: string, details: string): void {
        if (!this.m_instance) throw new Error("Logger not started on application load.");
        this.m_instance.m_logger.publish(logCode, message, origin, details);
    }
}
